### ISSUE TITLE
Linux Consumption Plan Auth Redirect

### ISSUE THREAD

**Issue by philtweir** [Azure/azure-functions-host#6532]
Linux cold starts on consumption plan have begun redirecting to AD login on an OPTIONS request, in at least certain circumstances (all cases we have tried). The failed request does not seem to trigger function start-up. Once a direct request is done to e.g. a GET endpoint, this succeeds as normal (and triggers function start) - further calls to OPTIONS and GET work as expected. This occurred on both our Python-based Linux consumption plan function apps. Upgrading to non-consumption plan resolved it, but required recreating the function apps.

#### Investigative information

Please provide the following:

- Timestamp: c. 2020-08-14 09:00:00 (first noticed 2020-08-13)
- Function App version: Python v2
- Invocation ID: (now deleted)
- Region: us-west-2

#### Repro steps

1. Create a Python function app under a Linux service plan as Dynamic-Y1 (via terraform)
2. Allow the function app to go cold
3. Do an OPTIONS request (such as a CORS preflight check)

#### Expected behavior

 - The OPTIONS request should wait for the app to start and return 200

#### Actual behavior

Provide a description of the actual behavior observed.

 - The OPTIONS request redirects to Azure AD

#### Known workarounds

 - Switch to a non-consumption plan, e.g. Standard-S1
 - Keep-alive the function with periodic requests
 - Hit the function with a GET before and OPTIONS request

#### Related information 

As Linux Consumption Apps do not allow easyauth, response (when cold) was instantaneous, there is no redirect in the function codebase, once warm the function worked correctly and the redirect was to the user's AD tenant, this seems to be an issue at the Azure end. Once the function app was working, via a direct call to a GET endpoint fired it up, then the OPTIONS request would succeed as normal, and continue working for around 30 min after the last query (not tested precisely). This appeared on a platform during a period of no changes from our side, and timing of appearance suggests it may be related to e.g. https://github.com/Azure/azure-functions-host/pull/6457

As CORS pre-flight was consequently not clearing, this prevented browsers from reaching the API entirely.

Additional information (may not be relevant):
  - Was being tested from Chrome and Firefox in a SPA app
  - Users were registered in AD B2C
  - A bearer token is included with requests (and is consumed within the app)
  - No record of the request appears in the Function App logs or metrics in the Azure portal

**Comment by balag0** [comment]
Please share the app name. You can use https://github.com/Azure/azure-functions-host/wiki/Sharing-Your-Function-App-name-privately if required.

Is EasyAuth enabled on the function app? Does turning that off help?

cc @ConnorMcMahon

**Comment by philtweir** [comment]
No EasyAuth - it is not possible to enable EasyAuth on a Linux Consumption Plan app -- this is a separate issue, but seems to be by design: https://github.com/MicrosoftDocs/azure-docs/issues/52954

I'd rather not share the app name, but as it was deleted (as part of the workaround) I don't have an invocation ID either -- I would need to create fresh again on a new Linux Service Plan.

**Comment by balag0** [comment]
Got it. I was not sure what to make of 
_Users were registered in AD B2C_

Also it looked like you mentioned the app redirects to AD tenant so not sure where that is coming if from if there is no EasyAuth enabled.

Anyway if you can share a repro or appname which is showing this behaviour we can investigate.

**Comment by philtweir** [comment]
Great - thanks. Yes, I was unclear on where that was coming from also - other than the JWT, I'm not too sure where it'd be getting tenant details.

I don't see any reason that using AD B2C (vs usual AD) would be relevant, but we have hit one or two issues with B2C/Function interaction that have been unexpected (e.g. https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/1795)

**Comment by v-anvari** [comment]
Hi @philtweir , Apologies for the delayed response. Were you able find a solution? do let us know if a follow up is required

**Comment by philtweir** [comment]
Yes, thanks @v-anvari - we switched to Kubernetes.

I hope the detailed info on the faulty PR was helpful to your team nonetheless!

**Comment by v-anvari** [comment]
Thank you for the confirmation. please feel free to reach back to us or re-open this issue in case of any queries.
